File: antlr.js
const antlr4 = require('antlr4/index');
const Lexer = require('./VectorsLexer');
const Parser = require('./VectorsParser');
const MainListener = require('./MainListener').MainListener;
 
let input = "vector p=[1,-2,3.2],p3;print p3;p3=[1,2,4];p3=p;print p3;";
let chars = new antlr4.InputStream(input);
let lexer = new Lexer.VectorsLexer(chars);
let tokens  = new antlr4.CommonTokenStream(lexer);
let parser = new Parser.VectorsParser(tokens);
parser.buildParseTrees = true;
let tree = parser.vectors();
let vec = new MainListener();
antlr4.tree.ParseTreeWalker.DEFAULT.walk(vec, tree);

File: antlr.py
from antlr4 import *
from VectorsLexer import VectorsLexer
from VectorsParser import VectorsParser
from MainListener import MainListener

def main():
    input = FileStream('./test.txt')
    lexer = VectorsLexer(input)
    stream = CommonTokenStream(lexer)
    parser = VectorsParser(stream)
    tree = parser.vectors()

    vec = MainListener()
    walker = ParseTreeWalker()
    walker.walk(vec, tree)

if __name__ == '__main__':
    main()
